lib/ansible/module_utils/aws.py

```diff
@@ -0,0 +1,142 @@
+**is identified.  In particular, until this notice goes away or is**
+**changed, methods may disappear from the interface.  Please don't**
+**publish modules using this except directly to the main Ansible**
+**development repository.**
+
+In order to use this module, include it as part of a custom
+module as shown below.
+
+  from ansible.module_utils.aws import AnsibleAWSModule
+  module = AnsibleAWSModule(argument_spec=dictionary, supports_check_mode=boolean
+                            mutually_exclusive=list1, required_together=list2)
+
+The 'AnsibleAWSModule' module provides similar, but more restricted,
+interfaces to the normal Ansible module.  It also includes the following additional methods.
+
+  m.fail_json_aws(Exception) # - take an exception and make a decent failure
+  m.get_conn() # - get an AWS connection.
+
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from ansible.module_utils.ec2 import HAS_BOTO3, camel_dict_to_snake_dict, ec2_argument_spec
+import traceback
+
+# We will also export HAS_BOTO3 so end user modules can use it.
+HAS_BOTO3 = HAS_BOTO3
+
+
+class AnsibleAWSModule(object):
+    """An ansible module class for AWS modules
+
+    AnsibleAWSModule provides an a class for building modules which
+    connect to Amazon Web Services.  The interface is currently more
+    restricted than the basic module class with the aim that later the
+    basic module class can be reduced.  If you find that any key
+    feature is missing please contact the author/Ansible AWS team
+    (available on #ansible-aws on IRC) to request the additional
+    features needed.
+    """
+
+    def __init__(self, **kwargs):
+        try:
+            default_args = kwargs["default_args"]
+            del kwargs["default_args"]
+        except KeyError:
+            default_args = True
+
+        try:
+            check_boto3 = kwargs["check_boto3"]
+            del kwargs["check_boto3"]
+        except KeyError:
+            check_boto3 = True
+
+        # Suggest that we autoretry on various AWS calls unless told not to:
+
+        try:
+            autoretry = kwargs["autoretry"]
+            del kwargs["autoretry"]
+        except KeyError:
+            autoretry = True
+        self.autoretry = autoretry
+
+        if default_args:
+            argument_spec_full = ec2_argument_spec()
+            try:
+                argument_spec_full.update(argument_spec)
+            except (TypeError, NameError):
+                pass
+
+        self._module = AnsibleModule(**kwargs)
+
+        if check_boto3 and not HAS_BOTO3:
+            self._module.fail_json(
+                msg='Python modules "botocore" or "boto3" are missing, please install both')
+
+        self.check_mode = self._module.check_mode
+
+    @property
+    def params(self):
+        return self._module.params
+
+    def exit_json(self, *args, **kwargs):
+        return self._module.exit_json(*args, **kwargs)
```

Comment:
And that way we can add features to existing interfaces without users needing to change their code from referring to the module vs. the AWSModule.